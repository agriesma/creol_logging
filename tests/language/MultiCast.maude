load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "Test" : Class | Inh: noInh, Param: noVid, Att: noSubst,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "m" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
      Ocnt: 0 >
    
    < "Main" : Class | Inh: noInh, Param: noVid, Att: "l" |-> null,
      Mtds: < "init" : Method | Param: noVid, Att: "o" |-> null, "_" |->
              null, Code: assign( ( "l" @ "Main" ) ; list(emp) ) ;
              new( "o" ; "Test" ; emp ) ; assign( "l" ; "-|" ( "o" :: "l" ) )
              ; new( "o" ; "Test" ; emp ) ; assign( "l" ; "-|" ( "o" ::
              "l" ) ) ; return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null,
              Code: multicast( "l" ; "m" ; emp ) ; return ( emp ) >,
      Ocnt: 0 > } .
endm

