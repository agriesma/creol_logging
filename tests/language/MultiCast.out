==========================================
reduce in PROGRAM : state .
result State: {
< "Main" : Class | Inh: noInh, Param: noVid, Att: "l" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, "o" |-> null, Code:
    assign("l" @ "Main" ; list(emp)) ; new("o" ; "Test" ; emp) ; assign("l" ;
    "-|"("o" :: "l")) ; new("o" ; "Test" ; emp) ; assign("l" ; "-|"("o" ::
    "l")) ; return(emp) >, < "run" : Method | Param: noVid, Att: "_" |-> null,
    Code: multicast("l" ; "m" ; emp) ; return(emp) >,
  Ocnt: 0 >
 
< "Test" : Class | Inh: noInh, Param: noVid, Att: noSubst, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(emp)
    >, < "m" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(
    emp) >, < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
    return(emp) >,
  Ocnt: 0 >
}
Bye.
